### PR TITLE
feat(playground-v2): playground supported wallet section

### PR DESCRIPTION
**Pull request by samobasquiat**

### CODE REVIEW

**Comment by mikasackermn** [comment]
Screenshot from 2023-10-11 12-27-09


According to figma, there is no need to display all the selected items
Screenshot from 2023-10-11 12-28-27

**Comment by mikasackermn** [comment]
Screenshot from 2023-10-11 13-02-34
Please check this

File: widget/playground/src/components/SupportedWallets/SupportedWallets.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      ),
+      id: title,
+    };
+  });
+
+  return (
+    <>
+      <HeaderContainer>
+        <div className="header">
+          <WalletIcon size={16} />
```
[the remark this answers is not in the document]
Comment (samobasquiat) [reply widget/playground/src/components/SupportedWallets/SupportedWallets.tsx:94]: fixed.

File: widget/playground/src/components/MultiSelect/MultiSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          <div className="chips">
+            {!value ? (
+              <Chip label={`All ${type}`} />
+            ) : !value.length ? (
+              <Chip label="None Selected" />
+            ) : (
+              value.map((v) => <Chip key={v} label={v} />)
+            )}
+          </div>
+          <div onClick={onClick} className="icon-btn">
```
[the remark this answers is not in the document]
Comment (samobasquiat) [reply widget/playground/src/components/MultiSelect/MultiSelect.tsx:64]: fixed.

File: widget/playground/src/components/MultiSelect/MultiSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+export function MultiSelect(props: PropTypes) {
+  const { label, type, value, onClick, icon } = props;
+  return (
+    <>
+      <Label>
+        {icon}
+        <Divider direction="horizontal" size={4} />
+        <Typography size="medium" variant="body">
+          {label}
```
[the remark this answers is not in the document]
Comment (samobasquiat) [reply widget/playground/src/components/MultiSelect/MultiSelect.tsx:49]: fixed.

**Review by mikasackermn — APPROVED** [review]

File: widget/playground/src/containers/FunctionalLayout/FunctionalLayout.Wallets.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          It's a sample using metamask, You can use your own wallet or what we
+          alredy implemented, check it out here.
+        </Typography>
+        <Divider size={16} />
+        <div className="footer">
+          <Button
+            type={externalWallets ? 'primary' : 'secondary'}
+            size="small"
+            variant="outlined"
+            style={{
```
Comment (RanGojo) [inline widget/playground/src/containers/FunctionalLayout/FunctionalLayout.Wallets.tsx:105]: Please avoid inline styles.

File: widget/playground/src/helpers.ts [hunk 5]
```diff
@@ -224,3 +223,34 @@ export const getStateWallet = (state: {
       return WalletState.DISCONNECTED;
   }
 };
+
+export function getWalletNetworks(chains: BlockchainMeta[]) {
```
Comment (RanGojo) [inline widget/playground/src/helpers.ts:227]: You could probably refactor this code like this:


```
export function getWalletNetworks(chains: BlockchainMeta[]) {
  const supportedNetworks: Set<BlockchainCategories> = new Set();

  chains.forEach((chain) => {
    switch (chain.type) {
      case TransactionType.EVM:
        supportedNetworks.add(BlockchainCategories.EVM);
        break;
      case TransactionType.COSMOS:
        supportedNetworks.add(BlockchainCategories.COSMOS);
        break;
      case TransactionType.TRANSFER:
        supportedNetworks.add(BlockchainCategories.UTXO);
        break;
      default:
        supportedNetworks.add(BlockchainCategories.OTHER);
        break;
    }
  });

  return Array.from(supportedNetworks);
}
```

File: widget/playground/src/helpers.ts [hunk 6]
```diff
@@ -176,31 +175,31 @@ export function insertAt(
 }
 
 export function formatConfig(config: WidgetConfig) {
```
Comment (RanGojo) [inline widget/playground/src/helpers.ts:177]: Please check the export code component again. it seems your changes to this function (and this file) break the code indentation.

File: widget/playground/src/components/SupportedWallets/SupportedWallets.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        w.title.toLowerCase().includes(searchValue.toLowerCase())
+      )
+    : allWallets;
+  const walletsList =
+    category === 'ALL'
+      ? filteredList
+      : filteredList.filter((wallet) => wallet.networks.includes(category));
+  const onChangeWallet = (wallet: WalletTypes) => {
+    const list = [...selectedWallets];
+    const isAlreadyExist =
```
Comment (RanGojo) [inline widget/playground/src/components/SupportedWallets/SupportedWallets.tsx:51]: You could rewrite it like this:

`const isAlreadyExist = list.includes(wallet);`

**Comment by RanGojo** [comment]
- [ ] I think `Select all / Deselect all` should apply to each category for the wallet section (and also other parts)
- [ ] When I select all of the wallets, we should see initial state in exported config.
- [ ] It would be better if widget config for `wallets` and `externalWallets` come after `apiKey` and `walletConnectProjectId`.

File: widget/playground/src/components/SupportedWallets/SupportedWallets.tsx (hunk 7 above)
Comment (samobasquiat) [reply widget/playground/src/components/SupportedWallets/SupportedWallets.tsx:51]: fixed. I had used previous logic.

File: widget/playground/src/containers/FunctionalLayout/FunctionalLayout.Wallets.tsx (hunk 4 above)
Comment (samobasquiat) [reply widget/playground/src/containers/FunctionalLayout/FunctionalLayout.Wallets.tsx:105]: In our codes, paddings & some other styles don't override the main component styles and we should make it important!

**Comment by samobasquiat** [comment]
> * [ ]  I think `Select all / Deselect all` should apply to each category for the wallet section (and also other parts)
> * [x]  When I select all of the wallets, we should see initial state in exported config.
> * [ ]  It would be better if widget config for `wallets` and `externalWallets` come after `apiKey` and `walletConnectProjectId`.
> * [x]  Although it's not directly related to this task, it would be great if, while switching between functional and style tabs, we consider open state for the first collapsible section.



> * [ ]  I think `Select all / Deselect all` should apply to each category for the wallet section (and also other parts)
> * [x]  When I select all of the wallets, we should see initial state in exported config.
> * [ ]  It would be better if widget config for `wallets` and `externalWallets` come after `apiKey` and `walletConnectProjectId`.
> * [x]  Although it's not directly related to this task, it would be great if, while switching between functional and style tabs, we consider open state for the first collapsible section.

- The component should be more general. In other future PRs, the select/Deselect's behavior will change.
- about the order of keys of widget config, it is not in the scope of this task because we use a lib to stringify and should check it.

File: widget/playground/src/helpers.ts (hunk 6 above)
Comment (samobasquiat) [reply widget/playground/src/helpers.ts]: fixed.

File: widget/playground/src/containers/FunctionalLayout/FunctionalLayout.Wallets.tsx (hunk 4 above)
Comment (samobasquiat) [reply widget/playground/src/containers/FunctionalLayout/FunctionalLayout.Wallets.tsx]: I used the variants version to remove inline styles.

**Pull request merged**